pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -2267,6 +2249,28 @@ private void initDeadLetterProducerIfNeeded() {
         }
     }
 
+    private void initRetryLetterProducerIfNeeded() {
+        if (retryLetterProducer == null) {
+            createProducerLock.writeLock().lock();
+            try {
+                if (retryLetterProducer == null) {
+                    retryLetterProducer = client
+                            .newProducer(Schema.AUTO_PRODUCE_BYTES(schema))
+                            .topic(this.deadLetterPolicy.getRetryLetterTopic())
+                            .enableBatching(false)
+                            .enableChunking(true)
+                            .blockIfQueueFull(false)
+                            .createAsync();
+                    retryLetterProducer.thenAccept(rtlProducer -> {
+                        stats.setDeadLetterProducerStats(rtlProducer.getStats());
```

Comment:
There's a typo here which makes the test `SimpleProducerConsumerStatTest.testPartitionedRetryLetterAndDeadLetterStats` fail. 

```suggestion
                        stats.setRetryLetterProducerStats(rtlProducer.getStats());
```